path_role.go

```diff
@@ -426,8 +436,31 @@ func (b *jwtAuthBackend) pathRoleCreateUpdate(ctx context.Context, req *logical.
 		role.VerboseOIDCLogging = verboseOIDCLoggingRaw.(bool)
 	}
 
+	boundClaimsType := data.Get("bound_claims_type").(string)
+	if boundClaimsType != boundClaimsTypeString && boundClaimsType != boundClaimsTypeGlob {
+		return logical.ErrorResponse("invalid 'bound_claims_type': %s", boundClaimsType), nil
+	}
+	role.BoundClaimsType = boundClaimsType
+
 	if boundClaimsRaw, ok := data.GetOk("bound_claims"); ok {
 		role.BoundClaims = boundClaimsRaw.(map[string]interface{})
+
+		if boundClaimsType == boundClaimsTypeGlob {
+			// Check that the claims are all strings
+			for _, claimValues := range role.BoundClaims {
+				claimsValuesList, ok := normalizeList(claimValues)
+
+				if !ok {
+					return logical.ErrorResponse("received claim is not a string or list: %v", claimValues), nil
+				}
+
+				for _, claimValue := range claimsValuesList {
+					if _, ok := claimValue.(string); !ok {
+						return logical.ErrorResponse("received claim is not a string: %v", claimValue), nil
```

Comment:
```suggestion
						return logical.ErrorResponse("claim is not a string: %v", claimValue), nil
```

Reply:
Removed the check in the new commit (to simply reviewing it)